homeassistant/components/sms/notify.py

```diff
@@ -20,31 +20,32 @@
 def get_service(hass, config, discovery_info=None):
     """Get the SMS notification service."""
 
-    if SMS_GATEWAY not in hass.data[DOMAIN]:
-        _LOGGER.error("SMS gateway not found, cannot initialize service")
-        return
-
-    gateway = hass.data[DOMAIN][SMS_GATEWAY][GATEWAY]
-
     if discovery_info is None:
         number = config[CONF_RECIPIENT]
     else:
         number = discovery_info[CONF_RECIPIENT]
 
-    return SMSNotificationService(gateway, number)
+    return SMSNotificationService(hass, number)
 
 
 class SMSNotificationService(BaseNotificationService):
     """Implement the notification service for SMS."""
 
-    def __init__(self, gateway, number):
+    def __init__(self, hass, number):
         """Initialize the service."""
-        self.gateway = gateway
+
+        self.hass = hass
         self.number = number
 
     async def async_send_message(self, message="", **kwargs):
         """Send SMS message."""
 
+        if SMS_GATEWAY not in self.hass.data[DOMAIN]:
```

Comment:
The correct fix is to load the notify platform via our discovery helper `async_load_platform` after the gateway is set up in `async_setup_entry` in the init module.

This notify platform should not be set up via a platform config.

Reply:
> The correct fix is to load the notify platform via our discovery helper `async_load_platform` after the gateway is set up in `async_setup_entry` in the init module.
> 
> This notify platform should not be set up via a platform config.

Thanks @MartinHjelmare . The current SMS notify is using platform config, but I would like to add a new one that does not require config. Can you point me to a sample notification that does not use config?

Reply:
The tibber integration is one example. :+1: